sefaria/model/version_state.py

```diff
@@ -20,78 +20,50 @@
-            'percentAvailable':        # duplicate
-            'isSparse':
-        }
-        "en":
-    }
-
-
-and now self.content is:
-    {
-        "_en": {
-            "availableTexts":
-            "availableCounts":
-            "percentAvailable":
-            "textComplete":
-            'completenessPercent':
-            'percentAvailableInvalid':
-            'sparseness':  # was isSparse
-        }
-        "_he": ...
-        "_all" {
-            "availableTexts":
-            "shape":
-                For depth 1: Integer - length
-                For depth 2: List of chapter lengths
-                For depth 3: List of list of chapter lengths?
-        }
-    }
-
 '''
 
 
 class VersionState(abst.AbstractMongoRecord, AbstractSchemaContent):
     """
     This model overrides default init/load/save behavior, since there is one and only one VersionState record for each Index record.
+
+    The `content` attribute is a dictionary which is the root of a tree, mirroring the shape of a Version, where the leaf nodes of the tree are dictionaries with a shape like the following::
+        {
+            "_en": {
+                "availableTexts":  Mask of what texts are available in this language.  Boolean values (0 or 1) in the shape of the JaggedArray
+                "availableCounts":  Array, with length == depth of the node.  Each element is the number of available elements at that depth.  e.g [chapters, verses]
+                "percentAvailable":  Percent of this text available in this language TODO: Only used on the dashboard. Remove?
+                'percentAvailableInvalid':  Boolean. Whether the value of "percentAvailable" can be trusted.  TODO: Only used on the dashboard. Remove?
+                "textComplete":  Boolean. Whether the text is complete in this language. TODO: Not used outside of this file. Should be removed.
+                'completenessPercent':  Percent of this text complete in this language TODO: Not used outside of this file. Should be removed.
+                'sparseness': Legacy - present on some records, but no longer in code TODO: remove
+            }
+            "_he": {...} # same keys as _en
+            "_all" {
+                "availableTexts": Mask what texts are available in this text overall.  Boolean values (0 or 1) in the shape of the JaggedArray
+                "shape":
+                    For depth 1: Integer -length
+                    For depth 2: List of section lengths
+                    For depth 3: List of list of section lengths
+            }
+        }
+
+    For example:
+    - the `content` attribute for a simple text like `Genesis` will be a dictionary with keys "_en", "_he", and "_all", as above.
+    - the `content` attribute for `Pesach Haggadah` will be a dictionary with keys: "Kadesh", "Urchatz", "Karpas" ... each with a value of a dictionary like the above.
+        The key "Magid" has a value of dictionary with keys "Ha Lachma Anya", etc.
```

Comment:
A thought about this line for clarity:

`The key "Magid" has a value of a dictionary, where each key is a different sub-section of Magid. The value for each key is a dictionary as detailed above, specific to each sub-section. So for example, one key will be "Ha Lachma Anya" and the value will be a dictionary, like the above, specific to the details of "Ha Lachma Anya". `

Perhaps also it would be worth adding a line that seems to encompass the overarching rule: `Every JaggedArrayNode has a corresponding vstate dictionary. So for complex texts, each leaf node (and leaf nodes by definition must be JaggedArrayNodes) has this corresponding dictionary`

Reply:
Incorporated in latest commit.